app/rust/src/parser/c32.rs

```diff
@@ -64,55 +66,71 @@ fn double_sha256_checksum(data: &mut [u8; SHA256_LEN]) {
 fn double_sha256_checksum(data: &mut [u8; SHA256_LEN]) {
     let mut output = [0u8; SHA256_LEN];
     // safe to unwrap as we are passing the right len
-    sha256(&data[..21], &mut output[..]).unwrap();
+    sha256(&data[..21], &mut output[..]).apdu_unwrap();
     data.copy_from_slice(output.as_ref());
     // safe to unwrap as we are passing the right len
-    sha256(&data[..], &mut output).unwrap();
-    data[20..24].copy_from_slice(&output[..4])
+    sha256(&data[..], &mut output).apdu_unwrap();
+    let d = data.get_mut(20..24).apdu_unwrap();
+    let o = output.get(..4).apdu_unwrap();
+    d.iter_mut().zip(o.iter()).for_each(|(d, o)| *d = *o);
```

Comment:
copy from slice missing?

Reply:
I was getting better results in binary size using them instead of copy_from_slice